src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -139,24 +141,35 @@ def make_dependency_files(parsed_config, config_file_path, to_stdout):
         for file_type in file_types_to_generate:
             for matrix_combo in grid(file_config["matrix"]):
                 dependencies = []
-                for dependency_entry in dependency_entries:
-                    if file_type not in get_entry_output_types(
-                        dependency_entry["output_types"]
-                    ):
-                        continue
-
-                    for include in includes:
-                        dependencies.extend(
-                            dependency_entry.get("common", {}).get(include, [])
-                        )
-
-                    for specific_entry in dependency_entry.get("specific", []):
-                        if not should_use_specific_entry(
-                            matrix_combo, specific_entry["matrix"]
-                        ):
+
+                for include in includes:
+                    dependency_entry = parsed_config["dependencies"][include]
+                    common_entries = dependency_entry.get("common", [])
+                    specific_entries = dependency_entry.get("specific", [])
+
+                    for common_entry in common_entries:
+                        if file_type not in common_entry["output_types"]:
+                            continue
+                        dependencies.extend(common_entry["packages"])
+
+                    for specific_entry in specific_entries:
+                        if file_type not in specific_entry["output_types"]:
                             continue
-                        for include in includes:
-                            dependencies.extend(specific_entry.get(include, []))
+                        specific_matrices = specific_entry["matrices"]
+
+                        for specific_matrices_entry in specific_matrices:
+                            if not should_use_specific_entry(
+                                matrix_combo, specific_matrices_entry["matrix"]
+                            ):
+                                continue
+                            dependencies.extend(
+                                specific_matrices_entry["packages"] or []
+                            )
+                            break
```

Comment:
This is a little simpler and doesn't require an explicit `continue`.
```suggestion
                            if should_use_specific_entry(
                                matrix_combo, specific_matrices_entry["matrix"]
                            ):
                                dependencies.extend(
                                    specific_matrices_entry["packages"] or []
                                )
                                break
```